### PR TITLE
refactor: migrate Banned Users to Virtua

### PR DESCRIPTION
**Pull request by srijnabhargav**
## Proposed changes 

This PR migrates the **Banned Users** contextual bar from `react-virtuoso` to **Virtua** while preserving the existing behavior as closely as possible.

### Changes included

* Replaces the direct `Virtuoso` implementation with a local `BannedUsersVirtualList`.
* Preserves the existing virtualization behavior without introducing new shared abstractions.
* Keeps the current scrollbar behavior and viewport configuration.
* Preserves `BannedUsersItem`, data fetching, pagination, loading, empty, and error states.
* Preserves the existing unban flow and user interactions.
* Preserves the current row structure and accessibility behavior as closely as possible.
* Adds focused unit tests covering:

  * Virtua rendering
  * Loading, empty, and error states
  * Infinite loading
  * Repeated bottom-scroll protection
  * Callback identity changes
  * Underfilled viewport loading

This migration is intentionally scoped to replacing the virtualization layer only, avoiding unrelated behavioral or architectural changes.

## Issue(s)

Part of the GSoC Virtua migration effort.

## Steps to test or reproduce

1. Open a room with banned users.
2. Open the **Banned Users** tab from the contextual bar.
3. Verify the banned users list renders correctly.
4. Scroll through the list and verify additional users are loaded as expected.
5. Verify the **Unban** action continues to work correctly.
6. Verify loading, empty, and error states behave as before.

## Further comments

Following the project's migration strategy, this implementation focuses on preserving the existing behavior while replacing only the underlying virtualization library.

A local `BannedUsersVirtualList` was introduced instead of extending the shared `PaginatedVirtualList` to minimize behavioral changes and keep the migration isolated to this component. This keeps the implementation small, easier to review, and avoids introducing shared abstractions before they are needed.

### Verification

* ✅ Targeted ESLint passed.
* ✅ Focused Jest passed (`11` tests, `5` snapshots).
* ✅ Type checking completed.

## Summary by CodeRabbit

* **New Features**
  * Updated the room banned users list to use a virtualized, scrollable view for better performance with long lists.
  * Improved “Load more” behavior to automatically trigger when users scroll near the bottom.
* **Bug Fixes**
  * Reduced duplicate load-more requests and improved reliability across edge cases (e.g., underfilled viewport, zero-sized viewport, and missing virtualization handle).
* **Tests**
  * Replaced prior smoke coverage with behavior-focused unit tests validating rendering and scroll/pagination logic.

### CODE REVIEW

File: apps/meteor/client/views/room/contextualBar/BannedUsers/BannedUsers.spec.tsx [hunk 1]
```diff
@@ -1,19 +1,237 @@
...
+		expect(screen.queryByTestId('banned-users-virtual-list')).not.toBeInTheDocument();
+		unmountLoading();
+
+		const { unmount: unmountEmpty } = renderBannedUsers({ bannedUsers: [] });
+		expect(screen.queryByTestId('banned-users-virtual-list')).not.toBeInTheDocument();
+		unmountEmpty();
+
+		renderBannedUsers({ error: new Error('Failed to load banned users'), bannedUsers: [] });
+		expect(screen.queryByTestId('banned-users-virtual-list')).not.toBeInTheDocument();
+	});
```
Comment (MartinSchoeler) [inline apps/meteor/client/views/room/contextualBar/BannedUsers/BannedUsers.spec.tsx]: I think we don't need these 2 tests, let's make sure the storybook has both examples of a populated and empty list, then we let the existing snapshots tests validate the structure.

File: apps/meteor/client/views/room/contextualBar/BannedUsers/BannedUsers.spec.tsx [hunk 2]
```diff
@@ -1,19 +1,237 @@
...
+		renderBannedUsers({ onLoadMore });
+
+		mockVirtualizerHandle.scrollOffset = 700;
+		fireEvent.scroll(screen.getByTestId('banned-users-virtual-list'));
+		await advanceDebouncedScroll();
+
+		expect(onLoadMore).not.toHaveBeenCalled();
+	});
+
+	it('does not call onLoadMore when the viewport has no measurable height', async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/client/views/room/contextualBar/BannedUsers/BannedUsers.spec.tsx]: This one is too niche of an edge case, we can drop it too

**Review by MartinSchoeler — DISMISSED** [review]
